common/djangoapps/third_party_auth/saml.py

```diff
@@ -41,10 +42,18 @@ def auth_url(self):
         if not self._config.enabled:
             log.error('SAML authentication is not enabled')
             raise Http404
+
         # TODO: remove this check once the fix is merged upstream:
         # https://github.com/omab/python-social-auth/pull/821
         if 'idp' not in self.strategy.request_data():
             raise AuthMissingParameter(self, 'idp')
+
+        from .models import SAMLProviderConfig
+        provider = SAMLProviderConfig.current(self.strategy.request_data()['idp'])
+        if not provider.enabled_for_current_site:
+            log.error('The SAML authentication provider is disabled for this site')
+            raise Http404
```

Comment:
Thanks, I added a test that should cover this code. The test suite can take quite a while to startup after first booting the VM (even with `--disable-migrations` and `--fast`, so my tests are still booting locally for me. Despite this, I pushed the changes and hopefully this test will pass/the coverage will increase to 100% for that file.